dataservice/api/biospecimen/schemas.py

```diff
@@ -49,5 +50,7 @@ class Meta(BaseSchema.Meta):
         'sequencing_center': ma.URLFor('api.sequencing_centers',
                                        kf_id='<sequencing_center_id>'),
         'genomic_files': ma.URLFor('api.genomic_files_list',
+                                   biospecimen_id='<kf_id>'),
+        'diagnoses': PatchedURLFor('api.diagnoses_list',
```

Comment:
I think you're ok using regular URLFor here since kf_id will never be null. PatchedURLFor is only when a foreign key could be null. The Biospecimen - Diagnosis relationship is exactly like Family - Participant. See [FamilySchema](https://github.com/kids-first/kf-api-dataservice/blob/27a4c0a73d4de1521fc50f62e5ec2de069519fa2/dataservice/api/family/schemas.py#L17) for reference.